backend/dataall/modules/worksheets/aws/glue_client.py

```diff
@@ -0,0 +1,36 @@
+import logging
+from os import name
+
+from botocore.exceptions import ClientError
+
+from dataall.base.aws.sts import SessionHelper
+
+log = logging.getLogger(__name__)
+
+
+class GlueClient:
+    def __init__(self, account_id, region, role=None):
+        pivot_role_session = SessionHelper.remote_session(accountid=account_id, region=region)
+        aws_session = (
+            SessionHelper.get_session(base_session=pivot_role_session, role_arn=role) if role else pivot_role_session
+        )
```

Comment:
yes - it is as such to be flexible for if there are other boto3 Glue APIs that need to be used at a later date that may not need to confine to this role chaining and just need pivot role

for the NLQ features this role will always be present because as you said we want to get the table metadata information as the team-IAM-role entity (or fail if we do not have access)